### ISSUE TITLE
Rule "one-var" with option "consecutive" should respect comments between declarations.

### ISSUE THREAD

**Issue by metaa** [eslint/eslint#10800]
**What rule do you want to change?**
*one-var*

**Does this change cause the rule to produce more or fewer warnings?**
Should be fewer, actually.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Adapt the behaviour of the `consecutive` option.

**Please provide some example code that this change will affect:**

Currently **incorrect**:
```js
/*eslint one-var: ["error", "consecutive"]*/
/*eslint-env es6*/

function example() {
	// This is very important
	let something = 'foo'

	// Will hold data about xyz
	let somethingElse

	// I don't even know
	let somethingCompletelyDifferent
}
```

**What does the rule currently do for this code?**
The `consecutive` option warns about consecutive declarations even if comments are inbetween.

**What will the rule do after it's changed?**
The `consecutive` option will ignore consecutive declaractions with comments inbetween, so that one has the possibility to declare variables separately and putting a comment about what it's used for etc. right above it.

**Comment by platinumazure** [comment]
I'm not sure about changing the default behavior. You can still use comments in this case:

```js
/*eslint one-var: ["error", "consecutive"]*/
/*eslint-env es6*/

function example() {
	// This is very important
	let something = 'foo',

	    // Will hold data about xyz
	    somethingElse,

	    // I don't even know
	    somethingCompletelyDifferent
}
```

So this feels like more of a stylistic preference than anything else.

I could get behind a new option though. Could be a string like `"consecutiveExceptComments"`, or could be an object option after the string option (`["consecutive", { allowCommentsBetweenDeclarations: true }]`), or something else.

**Comment by metaa** [comment]
Hmm, the format in your code block could be an ok solution, but I personally would go for an option to toggle 🙂
